x/ccv/consumer/keeper/genesis.go

```diff
@@ -97,63 +103,97 @@ func (k Keeper) InitGenesis(ctx sdk.Context, state *consumertypes.GenesisState)
+			mat := types.MaturingVSCPacket{
 				VscId:        vscId,
 				MaturityTime: timeNs,
 			}
 			maturingPackets = append(maturingPackets, mat)
 			return false
-		}
-		k.IteratePacketMaturityTime(ctx, cb)
+		})
 
-		gs.MaturingPackets = maturingPackets
-		return gs
-	}
-	clientID, ok := k.GetProviderClientID(ctx)
-	// if provider clientID and channelID don't exist on the consumer chain, then CCV protocol is disabled for this chain
-	// return a disabled genesis state
-	if !ok {
-		return consumertypes.DefaultGenesisState()
-	}
-	cs, ok := k.clientKeeper.GetClientState(ctx, clientID)
-	if !ok {
-		panic("provider client not set on already running consumer chain")
-	}
-	tmCs, ok := cs.(*ibctmtypes.ClientState)
-	if !ok {
-		panic("provider client consensus state is not tendermint client state")
-	}
-	consState, ok := k.clientKeeper.GetLatestClientConsensusState(ctx, clientID)
-	if !ok {
-		panic("provider consensus state not set on already running consumer chain")
-	}
-	tmConsState, ok := consState.(*ibctmtypes.ConsensusState)
-	if !ok {
-		panic("provider consensus state is not tendermint consensus state")
+		heightToVCIDs := []types.HeightToValsetUpdateID{}
+		k.IterateHeightToValsetUpdateID(ctx, func(height, vscID uint64) bool {
+			hv := types.HeightToValsetUpdateID{
+				Height:         height,
+				ValsetUpdateId: vscID,
+			}
+			heightToVCIDs = append(heightToVCIDs, hv)
+			return true
+		})
+
+		outstandingDowntimes := []types.OutstandingDowntime{}
+		k.IterateOutstandingDowntime(ctx, func(addr string) bool {
+			od := types.OutstandingDowntime{
+				ValidatorConsensusAddress: addr,
+			}
+			outstandingDowntimes = append(outstandingDowntimes, od)
+			return false
+		})
+
+		genesis = types.NewRestartGenesisState(
+			clientID,
+			channelID,
+			maturingPackets,
+			valset,
+			heightToVCIDs,
+			outstandingDowntimes,
+			params,
+		)
+	} else {
+		clientID, ok := k.GetProviderClientID(ctx)
+		// if provider clientID and channelID don't exist on the consumer chain, then CCV protocol is disabled for this chain
+		// return a disabled genesis state
+		if !ok {
+			return consumertypes.DefaultGenesisState()
+		}
+		cs, ok := k.clientKeeper.GetClientState(ctx, clientID)
+		if !ok {
+			panic("provider client not set on already running consumer chain")
+		}
+		tmCs, ok := cs.(*ibctmtypes.ClientState)
+		if !ok {
+			panic("provider client consensus state is not tendermint client state")
+		}
+		consState, ok := k.clientKeeper.GetLatestClientConsensusState(ctx, clientID)
+		if !ok {
+			panic("provider consensus state not set on already running consumer chain")
+		}
+		tmConsState, ok := consState.(*ibctmtypes.ConsensusState)
+		if !ok {
+			panic("provider consensus state is not tendermint consensus state")
+		}
+		// export client states and pending slashing requests into a new chain genesis
+		genesis = consumertypes.NewInitialGenesisState(tmCs, tmConsState, valset, k.GetPendingSlashRequests(ctx), params)
```

Comment:
Wouldn't a genesis state with `NewChain = true` lead to the creation of a new client in `InitGenesis` ([here](https://github.com/cosmos/interchain-security/blob/d62dc160b26627be2146db5db5a096a7672c4fa0/x/ccv/consumer/keeper/genesis.go#L41))? 

There are three states in which a consumer can start / restart:

1. A client to the provider was never created, i.e., this [line](https://github.com/cosmos/interchain-security/blob/d62dc160b26627be2146db5db5a096a7672c4fa0/x/ccv/consumer/keeper/genesis.go#L41) was never executed. 
2. The consumer restarts after the CCV channel was established, i.e., the check [here](https://github.com/cosmos/interchain-security/blob/d62dc160b26627be2146db5db5a096a7672c4fa0/x/ccv/consumer/keeper/genesis.go#L127) passes.
3. The client to the provider was created, but the channel handshake is still in progress. 

At the moment we miss the third case and we are trying to incorporate it in the first case.